test/moves/healing-wish-lunar-dance.test.ts

```diff
@@ -0,0 +1,247 @@
+import { GameManager } from "#test/test-utils/gameManager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it } from "vitest";
+
+describe("Moves - Lunar Dance and Healing Wish", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override.battleType("double").enemyAbility(AbilityId.BALL_FETCH).enemyMoveset(MoveId.SPLASH);
+  });
+
+  describe.each([
+    { moveName: "Healing Wish", moveId: MoveId.HEALING_WISH, restorePP: false },
+    { moveName: "Lunar Dance", moveId: MoveId.LUNAR_DANCE, restorePP: true },
+  ])("$moveName", ({ moveId, restorePP }) => {
+    it("should sacrifice the user to restore the switched in Pokemon's HP", async () => {
+      await game.classicMode.startBattle([SpeciesId.BULBASAUR, SpeciesId.CHARMANDER, SpeciesId.SQUIRTLE]);
+
+      const [bulbasaur, charmander, squirtle] = game.scene.getPlayerParty();
+      squirtle.hp = 1;
+
+      game.move.use(MoveId.SPLASH, 0);
+      game.move.use(moveId, 1);
+      game.selectPartyPokemon(2);
+
+      await game.toNextTurn();
+
+      expect(bulbasaur).toHaveFullHp();
+      expect(charmander.isFainted()).toBeTruthy();
+      expect(squirtle).toHaveFullHp();
+    });
+
+    it("should sacrifice the user to cure the switched in Pokemon's status", async () => {
+      game.override.statusEffect(StatusEffect.BURN);
+
+      await game.classicMode.startBattle([SpeciesId.BULBASAUR, SpeciesId.CHARMANDER, SpeciesId.SQUIRTLE]);
+      const [bulbasaur, charmander, squirtle] = game.scene.getPlayerParty();
+
+      game.move.use(MoveId.SPLASH, 0);
+      game.move.use(moveId, 1);
+      game.selectPartyPokemon(2);
+
+      await game.toNextTurn();
+
+      expect(bulbasaur).toHaveStatusEffect(StatusEffect.BURN);
+      expect(charmander.isFainted()).toBeTruthy();
+      expect(squirtle).toHaveStatusEffect(StatusEffect.NONE);
+    });
+
+    if (restorePP) {
+      it("should sacrifice the user to restore the switched in Pokemon's PP", async () => {
+        game.override.battleType("single");
+
+        await game.classicMode.startBattle([SpeciesId.BULBASAUR, SpeciesId.CHARMANDER]);
+        const [bulbasaur, charmander] = game.scene.getPlayerParty();
+        [bulbasaur, charmander].forEach((p) => game.move.changeMoveset(p, [moveId, MoveId.SPLASH]));
+
+        game.move.select(MoveId.SPLASH);
+        await game.toNextTurn();
+
+        game.switchPokemon(1);
+        await game.toNextTurn();
+
+        game.move.select(moveId);
+        game.selectPartyPokemon(1);
+
+        await game.toNextTurn();
+        expect(charmander.isFainted()).toBeTruthy();
+        bulbasaur.getMoveset().forEach((mv) => expect(mv.ppUsed).toBe(0));
+      });
+    }
+
```

Comment:
Since this test only affects Lunar-Dance you Probably want to move it outisde of the `describe.each()` instead of adding this `restorePP` logic cluster. That just adds unnecessary complexity